nway/nway_matching.py

```diff
@@ -133,13 +134,29 @@ def make_masks_from_dicts(self):
             path to json file containing nway matching input
 
         """
+        avg_proj_key = "ophys_average_intensity_projection_image"
         self.experiments = []
+        sizes = dict()
         for exp in self.args['experiment_containers']['ophys_experiments']:
             self.experiments.append(
                     utils.create_nice_mask(
                         exp,
                         self.args['output_directory']))
 
+            with PIL.Image.open(exp[avg_proj_key]) as im:
+                sizes.update({exp['id']: im.size})
+
+        try:
+            for a, b in itertools.combinations(sizes.values(), 2):
```

Comment:
How big to these dictionaries get? Would making all combinations ever be time consuming? Would it be worth doing something more like `x = np.array(sizes.values());    np.all(x == x[0])`?

Reply:
that would be another way, you're right, more efficient. But, the size is very small, maybe at-most, length 15 of a list like:
[(512, 447), (512, 447), ...]

Reply:
Oh ok Great! Then definitely ignore my suggestion, I think your way is a lot nicer when time isn't going to be a problem.

Reply:
without resorting to numpy, could do:
```
assert all([i == x[0] for i in x])
```

Reply:
Oh yeah that looks pretty nice and readable too, I like that

Reply:
done